### PR TITLE
feat(composite-product): copy composite product features into a new package

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
The composite product feature is included in the `@daffodil/product` package. This should be an optional feature.

Fixes: N/A

## What is the new behavior?
Added the `@daffodil/composite-product` subpackage and deprecated composite-related features in the main `@daffodil/product` package.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by lderrickable** [comment]
The README is not correct but everything else is ready for review

**Comment by lderrickable** [comment]
ready for review

File: libs/composite-product/tsconfig.lib.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "outDir": "../../dist/out-tsc/libs/product",
```
Comment (griest024) [inline libs/composite-product/tsconfig.lib.json:4]: I think this needs to be changed

**Review by griest024 — CHANGES_REQUESTED** [review]

File: libs/composite-product/tsconfig.lib.json (hunk 1 above)
Comment (lderrickable) [reply libs/composite-product/tsconfig.lib.json]: fixed

**Review by griest024 — COMMENTED** [review]
needs a rebase but lgtm otherwise

File: libs/composite-product/driver/magento/src/fragments/bundled-product.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import { gql } from 'apollo-angular';
+
+export const magentoBundledProductFragment = gql`
+  fragment magentoBundledProduct on BundleProduct {
+		items {
+			option_id
+			position
```
Comment (griest024) [inline libs/composite-product/driver/magento/src/fragments/bundled-product.ts:7]: `position` (and the one in the option) doesn't seem like they are ever used. If that's true then they should be removed.

**Review by griest024 — CHANGES_REQUESTED** [review]

File: libs/composite-product/driver/magento/src/fragments/bundled-product.ts (hunk 2 above)
Comment (lderrickable) [reply libs/composite-product/driver/magento/src/fragments/bundled-product.ts:7]: removed in this PR, then I'll rebase: https://github.com/graycoreio/daffodil/pull/1705

Comment (lderrickable) [reply libs/composite-product/driver/magento/src/fragments/bundled-product.ts]: rebased, and I made the same changes I made in the `@daffodil/product/driver/magento` package.

**Review by griest024 — APPROVED** [review]

**Review by griest024 — APPROVED** [review]

**Pull request merged**